fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -1429,11 +1429,13 @@ ACTOR Future<Void> assignMutationsToStorageServers(CommitBatchContext* self) {
 					double prob = mul * cost / totalCosts;
 
 					if (deterministicRandom()->random01() < prob) {
-						for (const auto& ssInfo : pProxyCommitData->keyInfo[m.param1].src_info) {
+						const auto& storageServers = pProxyCommitData->keyInfo[m.param1].src_info;
+						for (const auto& ssInfo : storageServers) {
 							auto id = ssInfo->interf.id();
 							// scale cost
 							cost = cost < CLIENT_KNOBS->COMMIT_SAMPLE_COST ? CLIENT_KNOBS->COMMIT_SAMPLE_COST : cost;
-							pProxyCommitData->updateSSTagCost(id, trs[self->transactionNum].tagSet.get(), m, cost);
```

Comment:
I wonder about the reason for averaging the committed cost by the size of source servers. In what test case you find it problematic? Because each SS needs to pay the cost of writing my instinct is to count them all.

Reply:
We already account for the write amplification in `getWriteOperationCost` with the `GLOBAL_TAG_THROTTLING_RW_FUNGIBILITY_RATIO` knob, so if we don't divide by `storageServers.size()` here, we'll overestimate the extra cost of writes.

Reply:
It might be worth keeping this in mind for the future; if/when we have dynamically scaling number of replicas, we may need to reduce the fungibility ratio (to something that assumes a single replica) and then account for replicas as needed.